jggtrans.8
.TH JGGTRANS 8 "$Date$"
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
jggtrans \- Jabber GaduGadu Transport
.SH SYNOPSIS
.B jggtrans
.RI [ options ]
.RI [<ConfigFile.xml>]
.SH DESCRIPTION
\fBjggtrans\fP is a gateway to GaduGadu instant messaging services, which is
very popular in Poland, made as an external component for Jabber server
daemon.
.SH OPTIONS
These programs follow the usual GNU command line syntax, without long
options.
A summary of options is included below.
.TP
.B \-h
Show summary of options.
.TP
.B \-f
Act on console (not as daemon), debug messages will be send to stderr.
.TP
.B \-d <n>
Set the debug level, 0 (default), >0 print more messages, <0 print less
messages.
.TP
.B \-D <n>
Set the debug level for libgadu (it also sets the
.B \-f
option.
.TP
.B \-u <user>
Switch to <user>s uid after start.
.TP
.B \-g <group>
Switch to <group>s gid after start.
.SH SEE ALSO
.BR jabberd (8)
.SH AUTHOR
This manual page was written by Robert Olejnik <robert@zaron.debian.pl>,
for the Debian project (but may be used by others).


NEWS
Version 2.0.5
-------------

This is minor feature improvement release.

Changes from 2.0.4:

* Fixed one presence bug ("GG" resource was not used everywhere as it should).

* All way of handling presence probes is back - no more errors instead
  of unavailable users.

* Jggtrans will now compile with TLS-enabled libgadu.

* TLS support (only with TLS-enabled libgadu).


Version 2.0.4
-------------

This is mainly bugfix/compatibility-improvement release.

Note: strange things may happen to GG contacts in users' rosters 
after upgrade until they log-off from Jabber and login again.

Changes from 2.0.3:

* All GG users use a "GG" resource.

* Search and registration form should now be fully compatible with JAJC's
  JEP-004 implementation.

* Support for Jabber presence type "invisible" added.

* Statistics gathering fixed, so multiple stats may be read at once.

Version 2.0.3
-------------

This is mainly bugfix release.

Changes from 2.0.2:

* Browsing support updated to the active JEP-11 version.

* More proxy configuration settings added: username, password and
  http_only flag.

Version 2.0.2
-------------

This is mainly bugfix release.

Changes from 2.0.1:

* Presence change notification handling improved. This should fix the
  annoying problem with transport users changing their GG presence
  multiple times when logging in.

* "get roster" doesn't add users to roster any more. There is new
  "import roster" command instead.

* There is an option in registration form allowing to auto-import
  userlist from GG server.

* Should now link on FreeBSD and other OSes without gettext in libc.

Version 2.0.1
-------------

This is mainly bugfix release.

Changes from 2.0.0:

* GG number used to register in transport is displayed with the "menu"
  and current settings

* Translation of local messages (log, --help) rewritten so it should
  really work now

* Fixes for bugs in "get roster" code, which caused transport crash or
  disconnection

* Other (minor) bugfixes


Version 2.0.0
-------------

This is a major feature improvement release. It may be not as stable as
1.4.x, but it is much cooler. However most of the new features are only
usable with modern-enough Jabber client.

Changes from 1.4.1:

* Userlist retrieval from GG server.

* Access Control Lists (ACL) may be defined in the configuration file.

* Statistic gathering (JEP-0039) support.

* Administrators (listed in the configuration file) can browse (using
  browse or disco) online users registered in the transport.

* Service discovery (JEP-0030) support.

* Jabber forms (data gathering and reporting, JEP-004) may be used for
  registration, searching and changing transport options.

* All messages generated by the transport (including error messages and
  logs content) maybe translated to user's or admin's favorite language.
  Polish translation is ready.

* GG public directory updates and password change (via the register
  form) work again.

* Several segmentation-fault bugs fixed.

* More GG client versions are now known to the transport.

Version 1.4.1
-------------

Changes from 1.4.0:

* autorestart (after connection to jabber server is broken) fixed.

* fix for automatic server change (now jggtrans should really change
  server, when one doesn't respond)

* searching by email removed (it seems this is not supported any more)

* fixed handling of delayed (stored on GG server) messages

* user settings, like invisible or friends-only, are properly saved now

* detection of new GG client versions

* command list is sent as one message now


-------------

Changes from 1.3.1:

* configurable GG server list.

* support for new GG search protocol. Searching and retrieving user
  information should work again.

* crash caused by invalid registration form fixed.


Version 1.3.1
-------------

Changes from 1.3.0:

* crash caused by empty presence priority (sometimes sent by Exodus) fixed

* pid file is written before privileges are dropped and is removed on exit


Version 1.3.0
-------------

Changes from 1.2.3:

* several (at least 3) segmentation faults fixed

* fixes for some "strange behaviour" in several Jabber clients 

* "friends only" and "invisible" modes (controlled by messages to the
  transport)

* support for user lists stored on GG server started (broken, disabled by
  default)

* restart after connection to Jabber server is broken

* GG client version detection

* pkg-config support for libgadu configure check

Version 1.2.3
-------------

Changes from 1.2.2:

* new character conversion function (instead of sometimes broken
  iconv()). Fixes crashes on invalid characters in GG input.

* jggtrans will now ignore control characters in GG input. These are
  invalid in XML stream
  
* fixes in contact status handling 

Not a change in the program, but important: jabber-gg-project is now
registered on JabberStudio (http://www.jabberstudio.org). This is the
proper place for bug reports and feature requests now.

Version 1.2.2
-------------

Changes from 1.2.1:

* two bugs in vCard and user directory fixed


Version 1.2.1
-------------

Changes from 1.2.0:

* compilation on FreeBSD fixed

* GG status change fixed when no reason string is given

* fixed freeing already free  xmlnode query  while registering.
  Patch by Tomasz Sterna <tomek@smoczy.net>

* added jabber:iq:version jggtrans query support. Psi nicely shows it on
  transport tooltip.
  Patch by Tomasz Sterna <tomek@smoczy.net>


Version 1.2.0
-------------

Changes from 1.1.0:

* updated to new libgadu (works with libgadu-20021205).

* searching works again.

* IP address of active GG user in roster is provided in "About" field of
  user's vCard.

* several memory leaks fixed.

* workaround for buggy support for jabber:iq:gateway in some clients.
  This fixes contact addition problems.


Version 1.1.0
-------------

Changes from 1.0.1:

* presence change reason (status description) support


Version 1.0.1
-------------

jggtrans.xml.Pl
<!--
       To polska wersja przykładowego pliku konfiguracyjnego.

       Proszę pamiętać, że w XMLu obowiązującym kodowaniem jest UTF-8
-->

<jggtrans>

  <!-- Żeby to zadziałało także plik jabber.xml musi być uaktualniony.
       Należy dodać następujące linie w sekcji <browse/>:
    
    <service type="x-gadugadu" jid="gg.localhost" name="GaduGadu Transport">
      <ns>jabber:iq:gateway</ns>
      <ns>jabber:iq:register</ns>
      <ns>jabber:iq:search</ns>
      <ns>jabber:iq:agent</ns>
      <ns>jabber:iq:browse</ns>
    </service>

  Oraz następujące, pod koniec pliku jabber.xml (poza <browse/>):

  <service id="gglinker">
    <host>gg.localhost</host>
    <accept>
      <ip>127.0.0.1</ip>
      <port>5353</port>
      <secret>secret</secret>
    </accept>
  </service>

  Oczywiście gg.localhost należy zmienić na własną nazwę serwisu (musi być w DNS),
  127.0.0.1 oraz 5353 na adres/port używany do komunikacji między jggtrans a jabberd,
  a secret na jakiś dowolny ciąg znaków.
  -->
   
  <!-- Ten wpis powinien zgadzać się z <service/> w sekcji <browse/> pliku jabber.xml -->
  <service jid="gg.localhost"/>

  <!-- A ten, z <service id="gglinker/> --> 
  <connect id="gglinker">
    <ip>127.0.0.1</ip>
    <port>5353</port>
    <secret>secret</secret>
  </connect>

  <register>
  	<!-- Tutaj zdefiniowany jest tekst informacji wyświetlanej 
	     użytkownikowi podczas rejestracji. 
	    
	     Do formatowania tego tekstu można użyć tagów: 
	     <p/> (nowy akapit) oraz <br/> (nowa linia).
	-->
  	<instructions>
		Aby się zarejestrować proszę wpisać 
		swój numerek GG (UIN) w polu "username".<p/>
		Aby się wyrejestrować należy zostawić formularz pusty.<p/>
		Aby zmienić swoje dane w bazie GaduGadu należy wypełnić 
		pole "nickname" oraz w razie potrzeby pozostałe.<p/>
		Wypełniając ten formularz zgadzasz się na wysłanie zamieszczonych
		tutaj danych do sms-express.com Sp. z o.o. i ich przetwarzanie
		przez tę spółkę. <p/>
		Na serwerze realizującym transport zachowane będzie jedynie hasło,
		jabberid oraz numer GG użytkownika, a także numery osób w liście kontaktów.
	</instructions>
  </register>
  
 <search>
  	<!-- Tutaj zdefiniowana jest instrukcja przeszukiwania dla użytkowników. -->
 	<instructions>
	  	Proszę podać imię (first) i/lub nazwisko (last) 
	  	i/lub ksywę (nick) i/lub miasto (city) szukanej osoby.<p/>
	 	Można także wyszukiwać według numeru UIN.
	</instructions>
 </search>

 <gateway>
  	<!-- 
	     Poniższy komunikat może być wyświetlony użytkownikowi 
	     przy dodawaniu kontaktu GG.                            
	-->
 	<desc>
		Proszę podać numerek GaduGadu użytkownika z którym chcesz się
		skontaktować.
	</desc>
	<prompt>
		Numerek GG
	</prompt>
 </gateway>

 <!-- Tutaj definiuje się "wizytówkę" (vCard) transportu. -->
 <vCard>
     <FN>GaduGadu Transport</FN>
     <DESC>To jest bramka pomiędzy Jabberem, a GaduGadu.</DESC>
     <URL>http://foo.bar/</URL>
 </vCard>

 <!-- domyślne locale (język) użytkownika. 
      Puste oznacza lokale systemowe, a brak to brak tłumaczeń -->
 <default_locale>pl_PL</default_locale>

 <!-- 
      Konfiguracja logów. 
      Można skonfigurować jeden logger typu "syslog" i/lub jeden "file".
      Można też nie skonfigurować żadnego logowania. -->
 <log type="syslog" facility="local0"/>
 <log type="file">/tmp/ggtrans.log</log>

 <!-- Odkomentuj to, jeśli chcesz, by do połączeń z serwerem GG było wykorzystane proxy -->
 <!--
 <proxy>
        <ip>127.0.0.1</ip>
        <port>8080</port>
 </proxy>
 -->
 
 <!-- Te wartości można zmieniać wedle własnych upodobań.
      Jak sesja niepotrzebnie dostaje timeout można to zwiększać 
      w nieskończoność :-) -->
 <conn_timeout>60</conn_timeout>
 <ping_interval>10</ping_interval>

 <!-- Od jakiegoś czasu serwer GaduGadu nie odpowiada na pingi :-( 
      Ale dajmy mu rok czasu na odpowiedź :-) -->
 <pong_timeout>31536000</pong_timeout>

 <!-- jak szybko ma transport spróbować się połączyć po rozłączeniu -->
 <reconnect>300</reconnect>

 <!-- Katalog z danymi użytkowników. -->
 <!-- Proszę uważać na uprawnienie - hasła użytkowników (do GG) będą tam przechowywane -->
 <spool>/var/lib/jabber/spool/gg.localhost/</spool>

 <!-- Ile czekać po przerwaniu połączenia z serwerem jabbera, zanim jggtrans
      się zrestartuje. Wartoś ujemna oznacza, że jggtrans ma w takim przypadku
      przerwać pracę. -->
 <restart_timeout>60</restart_timeout>

 <!-- lista serwerów GG których jggtrans ma używać 
 	<hub/> oznacza "użyj huba GG aby znaleźć właściwy serwer" 
      	Element <server/> powinien zawierać adres serwera. Może też mieć 
	ustawiony atrybut "port" zawierający numer portu na serwerze.
	Jggtrans będzie się próbował łączyć z podanymi serwerami w podanej
	kolejności i użyje pierwszego działającego.
	Rozsądna domyślna lista serwerów jest wkompilowana w jggtrans.
      -->
 <!--
 <servers>
	<hub/>
	<server>217.17.41.84</server> 
	<server>217.17.41.85</server> 
	<server port="8074">217.17.41.86</server> 
 </servers>
 -->
 
 <!-- Gdzie zapisywać/sprawdzać plik z pid procesu ggtrans. -->
 <pidfile>/var/lib/jabber/ggtrans.pid</pidfile>

</jggtrans>
<!--
 vi: encoding=utf-8 syntax=xml
 -->
